src/lib/transition_frontier/snark_pool_refcount.ml

```diff
@@ -16,17 +16,23 @@ module type S = Transition_frontier_extension_intf
 
 module Make (Inputs : Inputs_intf) :
   S
-  with type transition_frontier_breadcrumb := Inputs.Breadcrumb.t
-   and type input := unit
+  with type transition_frontier_breadcrumb = Inputs.Breadcrumb.t
+   and type input = unit
```

Comment:
can you change these back to `:=`

Reply:
If you use `:=` it substitutes the types out and the module no longer matches `Transition_frontier_extension_intf`, which means you have to duplicate the signature in `coda_transition_frontier.ml`.

Reply:
The downside is you have to repeat yourself in the implementation and merlin no longer gives you proper context on types when you ask. It becomes hard to trace the underlying types for things if you don't use substitution as the application gets more complex (this is why we settled on preferring the `:=` style for now). Since this extension interface is repeated with different specializations on `view` and `input`, I think you could make the argument that it should not use substitution.

To avoid repeating yourself on the parts that should be `:=` (the breadcrumb in this case), you can do the following:

```ocaml
module Transition_frontier_extension_intf (Inputs : Inputs_intf) = struct
   module type S = Transition_frontier_extension_intf0 (* the one we have right now *)
       with type transition_frontier_breadcrumb := Inputs.Breadcrumb.t
end 
```

and then you could specialize just the parameters that should be unique for the extension (and I believe your argument that maybe these shouldn't be substituted since they're unique for each extension):

```ocaml
module Make (Inputs: Inputs_intf) : 
	Transition_frontier_extension_intf(Inputs).S
		with type input = unit
         and type view = unit
```